src/deployment/azuredeploy.json

```diff
@@ -583,6 +538,46 @@
                 }
             ]
         },
+
+        {
+            "apiVersion": "2020-02-02-preview",
+            "name": "[parameters('name')]",
```

Comment:
This is the name of the App Insights resource, right? How about we similarly use a `variable()` here and in all the places where we use `parameters('name')` or `[resourceId('microsoft.insights/components', parameters('name'))]` to refer to it.

Reply:
@ranweiler, I ended up adding new variables for all of the resources name using parameters('name') and updating the references.